src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -163,6 +165,9 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
             _workspaceToLspSolution[workspace] = null;
         }
 
+        // Send a workspace changed notification to anyone subscribed. For example, this is important for semantic tokens refresh.
+        LspWorkspaceChanged?.Invoke(sender, e);
```

Comment:
Will the LSP workspace get updated on reference changes? Aka if there's no didChange or didOpen etc. will this fire?

Reply:
I'm pretty sure it will - the `OnWorkspaceChanged` event handler is subscribed to `Workspace.WorkspaceChanged` which I believe is separate from LSP

Reply:
Yeah workspace change encompasses any change to the workspace.  In the code above we ignore text changes to LSP documents, but everything else should trigger this